v2/coreComponentUtils/util.go

```diff
@@ -0,0 +1,75 @@
+package coreComponentUtils
+
+import (
+	"fmt"
+	"github.com/kubeflow/pipelines/api/v2alpha1/go/pipelinespec"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+func PrepareOutputFolders(executorInput *pipelinespec.ExecutorInput) error {
+	for name, parameter := range executorInput.GetOutputs().GetParameters() {
+		dir := filepath.Dir(parameter.OutputFile)
+		if err := os.MkdirAll(dir, 0644); err != nil {
+			return fmt.Errorf("failed to create directory %q for output parameter %q: %w", dir, name, err)
+		}
+	}
+
+	for name, artifactList := range executorInput.GetOutputs().GetArtifacts() {
+		if len(artifactList.Artifacts) == 0 {
+			continue
+		}
+		outputArtifact := artifactList.Artifacts[0]
+
+		localPath, err := LocalPathForURI(outputArtifact.Uri)
+		if err != nil {
+			return fmt.Errorf("failed to generate local storage path for output artifact %q: %w", name, err)
+		}
+
+		if err := os.MkdirAll(filepath.Dir(localPath), 0644); err != nil {
+			return fmt.Errorf("unable to create directory %q for output artifact %q: %w", filepath.Dir(localPath), name, err)
+		}
+	}
+
+	return nil
+}
+
+func LocalPathForURI(uri string) (string, error) {
+	if strings.HasPrefix(uri, "gs://") {
+		return "/gcs/" + strings.TrimPrefix(uri, "gs://"), nil
+	}
+	if strings.HasPrefix(uri, "minio://") {
+		return "/minio/" + strings.TrimPrefix(uri, "minio://"), nil
+	}
+	if strings.HasPrefix(uri, "s3://") {
+		return "/s3/" + strings.TrimPrefix(uri, "s3://"), nil
+	}
+	return "", fmt.Errorf("failed to generate local path for URI %s: unsupported storage scheme", uri)
+}
+
+// Add outputs info from component spec to executor input.
+func AddOutputs(executorInput *pipelinespec.ExecutorInput, outputs *pipelinespec.ComponentOutputsSpec) error {
```

Comment:
I haven't checked details, if this really needs to be shared, I recommend one of
* Make this a public method of component package and let driver import component.
* Merge driver package into component package. They don't seem very different anyway.

Reply:
I temporarily move shared constant to component package. I will later create another PR to move driver package into component package